src/Extensions/TimedScopes/TimedScopeProvider.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.Logging.TimedScopes
+{
+	internal class TimedScopeProvider : ITimedScopeProvider
+	{
+		public TimedScopeProvider(IMachineInformation machineInformation, TimedScopeEventSource eventSource)
+		{
+			m_serviceName = machineInformation.ServiceName;
+			m_eventSource = eventSource;
+		}
+
+		public TimedScope Start(string name, TimedScopeResult result) =>
+			new TimedScope(m_eventSource, m_serviceName, name, result);
+
+		private readonly TimedScopeEventSource m_eventSource;
+		private readonly string m_serviceName;
+	}
+
+
+	/// <summary>Interface to create TimedScope</summary>
+	public interface ITimedScopeProvider
```

Comment:
>ITimedScopeProvider [](start = 18, length = 19)

Please keep one class/interface per file. It's much easier to find things that way

Reply:
Fixed, thank you